internal/kms/kms.go

```diff
@@ -269,6 +271,43 @@ func (k *Kms) loadRoot(ctx context.Context, scopeId string, opt ...Option) (*mul
 	return multi, rootKeyId, nil
 }
 
+// ReconcileKeys will reconcile the keys in the kms against known possible issues.
+func (k *Kms) ReconcileKeys(ctx context.Context, randomReader io.Reader) error {
+	const op = "kms.ReconcileKeys"
+	if isNil(randomReader) {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing rand reader")
+	}
+
+	// it's possible that the global audit key was created after this instance's
+	// database was initialized... so check if the audit wrapper is available
+	// for the global scope and if not, then add one to the global scope
+	if _, err := k.GetWrapper(ctx, scope.Global.String(), KeyPurposeAudit); err != nil {
```

Comment:
It'd be better if there was a way of ensuring that the error is because the key doesn't exist rather than something else going wrong, e.g. switching on error type.

Reply:
ty!